Updates to Single Post spacing and styling

jordesign opened the pull request:
**Description**

Addressing #319 - this aims to update the single.html file to be closer to the [Figma design](https://www.figma.com/file/AlYr03vh4dVimwYwQkTdf6/Twenty-Twenty-Four?type=design&node-id=492-1457&mode=design&t=1ksmbRV1pC17uorE-0). Most notable additions.

* Update spacing between main elements (Title, Body, Comments).
* Add separator at bottom of post.
* Add CSS styling for the taxonomy/tags

**Screenshots**

![theme-building local_3-years-at-automattic_ (1)](https://github.com/WordPress/twentytwentyfour/assets/1842363/81368564-9f18-4302-889b-a9ecef1d0505)

** Questions **

I did my best to use theme presets where possible but there were a few aspects I needed to clarify..

* Comments title sizing was different - but uses the closest preset sizing to what we have.
* There's no preset colors with the subtlety used for the tags/taxonomy background - so I've hard-coded that into the CSS (which I imagine we need to amend).
* Similarly - the other values in the CSS for the tags didn't have matching presets

jordesign committed:
Update spacing in top of post

jordesign committed:
Tidy spacing at bottom of post

jordesign committed:
CSS styles for tags

jordesign committed:
Add bottom separator above comments

jordesign committed:
Tidy up spacing above separator

jordesign committed:
Make sure tag styling is specific to tags

carolinan commented:
All content needs to be inside a landmark.
The easiest way to solve this is to remove the current `<main>` and wrap everything that is between the header and footer in a new group.

MaggieCabrera commented on style.css:
```diff
@@ -52,3 +52,20 @@ a {
+ * https://github.com/WordPress/twentytwentyfour/issues/XX
+ */
+
+.wp-block-post-terms.taxonomy-post_tag a {
```
Let's make this a block style so users can turn it off

jordesign replied:
This is now using a block style 👍

MaggieCabrera reviewed (changes requested):
With the block style and the landmark issues resolved, this should be good to go

jordesign committed:
Add Block Style variation for post terms

jordesign committed:
Add CSS for style variation

It was tricky to have the style *not* show for the placeholders for prefix and suffix on the post-terms block.

jordesign committed:
Tidy up specificity of CSS selectors

jordesign committed:
Add Landmark for content post title/meta/content

jordesign commented:
@MaggieCabrera Post Terms are now using a block style, and I _think_ I've correctly included the main content area in a landmark (but TBH it's the first time I've tried that - so I'm open to feedback).

I think those changes might have interfered with the review you had of some of the spacing in the template so it might need a quick re-review I'm afraid.

MaggieCabrera committed:
renamed block style and tweaked the css

MaggieCabrera commented on style.css:
```diff
@@ -52,3 +52,22 @@ a {
+ */
+
+.is-style-pill a,
+.is-style-pill span:not([class], [data-rich-text-placeholder])
```
The prefix and suffix don't need the pill style, that's for when you want to add some text before or after the tag list, so I removed the styling for those

jordesign replied:
Ah awesome - thank you. I'd been trying to figure out how to not have it apply to the placeholder, but couldn't figure it out.

Also - Pill is a much better style name 😂

MaggieCabrera reviewed (approved):
looks good, thank you!

jordesign commented:
Thanks @MaggieCabrera  - with that styling confirmed for the single post - should it also be rolled across to the 'Single with sidebar' and 'Write single' templates?

MaggieCabrera commented:
> Thanks @MaggieCabrera - with that styling confirmed for the single post - should it also be rolled across to the 'Single with sidebar' and 'Write single' templates?

yeah, we can do that but I think Single writer needs to be removed, it's almost exactly like single

jordesign commented:
Agreed - it is essentially the same except the Figma design seems to not include the featured image. I'll add an issue/PR to remove it.

jordesign commented:
PR up in #349

jordesign commented:
[Issue](https://github.com/WordPress/twentytwentyfour/issues/395) and [PR](https://github.com/WordPress/twentytwentyfour/pull/396)
